### PR TITLE
[SPARK-24698][PYTHON]: Fixed typo in pyspark.ml's Identifiable class.

### PR DESCRIPTION
**Pull request by mcteo**
## What changes were proposed in this pull request?

Fixed a small typo in the code that caused 20 random characters to be added to the UID, rather than 12.

### CODE REVIEW

**Review by felixcheung — APPROVED** [review]
@HyukjinKwon @ueshin

**Comment by felixcheung** [comment]
ok to test

File: python/pyspark/ml/util.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def _randomUID(cls):
         Generate a unique unicode id for the object. The default implementation
         concatenates the class name, "_", and 12 random hex chars.
         """
-        return unicode(cls.__name__ + "_" + uuid.uuid4().hex[12:])
+        return unicode(cls.__name__ + "_" + uuid.uuid4().hex[:12])
```
Comment (viirya) [inline python/pyspark/ml/util.py]: `uuid.uuid4().hex[-12:]`? In Scala side, it is `UUID.randomUUID().toString.takeRight(12)`.

Comment (HyukjinKwon) [reply python/pyspark/ml/util.py]: Yea, `takeRight(12)` should be equivalent with `hex[-12:]`.

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
ping @mcteo mind addressing https://github.com/apache/spark/pull/21675#discussion_r199316192? I think it's good to address before merging it in.

**Comment by mcteo** [comment]
Hey @HyukjinKwon, made that change.

**Comment by HyukjinKwon** [comment]
Merged to master

**Pull request closed without merging**